_studio/shared/src/cm_mem_copy.cpp

```diff
@@ -3164,6 +3164,72 @@ mfxStatus CmCopyWrapper::CopyMirrorVideoToVideoMemory(void *pDst, void *pSrc, mf
 
 #define CM_RGB_SUPPORTED_COPY_SIZE(ROI) (ROI.width <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT && ROI.height <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT )
 
+bool CmCopyWrapper::CheckSurfaceContinuouslyAllocated(mfxFrameSurface1 &surf)
+{
+    mfxFrameData* ptrFrameData = &surf.Data;
+
+    switch (surf.Info.FourCC)
+    {
+    // Packed formats like YUY2, UYVY, AYUV, Y416, Y210, Y410, A2RGB10, RGB565, RGB3 
+    // does not need to be handled here - they got only one plane with all the luma and chroma data interleaved
+
+    // Handling semi-planar formats
+    case MFX_FOURCC_NV12:
+    case MFX_FOURCC_NV16:
+        {
+            mfxU32 stride_in_bytes = ptrFrameData->PitchLow + ((mfxU32)ptrFrameData->PitchHigh << 16);
+            size_t Y_size = stride_in_bytes * mfx::align2_value(surf.Info.Height, 32);
+            if (ptrFrameData->Y + Y_size + 1 != ptrFrameData->UV)
```

Comment:
What for +1 ?

Reply:
Yes, it is not needed, you've already skipped entire plane